components/locid_transform/src/fallback/mod.rs

```diff
@@ -386,22 +380,9 @@ impl LocaleFallbacker {
 }
 
 impl<'a> LocaleFallbackerBorrowed<'a> {
-    /// Creates an iterator based on a [`Locale`].
-    ///
-    /// When first initialized, the locale is normalized according to the fallback algorithm.
-    ///
-    /// For performance reasons, internally an [`icu_provider::DataLocale`] is used.
-    ///
-    /// [`Locale`]: icu_locid::Locale
```

Comment:
Why did you change this?

Reply:
You just noticed? :) because we use DataLocale basically everywhere, in all our constructors, and this is a place where we actually need it. As discussed previously, I dislike `Into` bounds because it creates generics at call sites when people should really just call `.into()` for the same behavior.

Reply:
I'm not talking about the general concept of using `DataLocale` here, I'm asking why you removed the docs that explain that it's fine if you have a `Locale`, but we use `DataLocale` for performance reasons.

Reply:
Discussed and revised docs.